### PR TITLE
add time-series example notebook

### PR DESCRIPTION
**Pull request by zicanl-amazon**
*Issue #, if available:*

*Description of changes: Add examples for TimeSeries Explainability

*Testing done: Notebook runs end to end

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,6 +205,7 @@ These examples provide an Introduction to Smart Sifting library. Smart Sifting i
 These examples provide an introduction to SageMaker Clarify which provides machine learning developers with greater visibility into their training data and models so they can identify and limit bias and explain predictions.
 
 * [Fairness and Explainability with SageMaker Clarify](sagemaker-clarify/fairness_and_explainability) shows how to use SageMaker Clarify Processor API to measure the pre-training bias of a dataset and post-training bias of a model, and explain the importance of the input features on the model's decision.
+* [TimeSeries Explainability with SageMaker Clarify](sagemaker-clarify/time_series_deepar) shows how to use SageMaker Clarify Processor API to explain the importance of the input features on the time-series model's decision.
 * [Amazon SageMaker Clarify Model Monitors](sagemaker_model_monitor/fairness_and_explainability) shows how to use SageMaker Clarify Model Monitor API to schedule bias monitor to monitor predictions for bias drift on a regular basis, and schedule explainability monitor to monitor predictions for feature attribution drift on a regular basis.
```
[the remark this answers is not in the document]
Comment (aws-byeldos) [reply README.md:210]: Question:

In the "Display Results", should we output only a couple of lines outputs, so the page is not huge to load?

[the remark this answers is not in the document]
Comment (zicanl-amazon) [reply README.md:210]: 1. Will do. 
2. We already use Mercury to improve the layout. It won't be too huge in the jupyter notebook
3. Notebook CI Test Result is the CI badge required in the CONTRIBUTING.me: https://github.com/aws/amazon-sagemaker-examples/blob/main/CONTRIBUTING.md#add-ci-badges-to-your-notebook

**Review by matrix-compute — APPROVED** [review]

**Review by rvasahu-amazon — APPROVED** [review]
Please implement the applicable feedback in the failing notebook grammar test.

File: sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb [hunk 2]
```diff
@@ -0,0 +1,1019 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "4a6b3a4e",
+   "metadata": {},
+   "source": [
+    "# TimeSeries Bing Your Own Model"
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:8]: Currently says "Bing" Your Own Model, change to "Bring" please.

Also maybe put a space in Time Series.

File: sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb [hunk 3]
```diff
@@ -0,0 +1,1019 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1686b70e",
+   "metadata": {},
+   "source": [
+    "## Install Development SDK\n",
+    "\n",
+    "As of 13-12-23, time series explainability (TSX) is in closed beta. The changes to the SDK to support it are not merged into the `sagemaker-python-sdk` repository. Instead, we install a version of the PySDK that does support these changes directly via a wheel file. "
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:85]: The notebook no longer installs the SDK from a wheel file, please update this description.

File: sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb [hunk 4]
```diff
@@ -0,0 +1,1019 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b7597402",
+   "metadata": {},
+   "source": [
+    "### Import Libraries\n",
+    "\n",
+    "The model used in this example notebook is DeepAr forecasting model in AWS. For more information, please check https://docs.aws.amazon.com/sagemaker/latest/dg/deepar.html (TODO, change this)\n",
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:198]: This notebook is for the BYOM demo, correct? Description says this uses the DeepAR forecasting model. Please update.

File: sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb (hunk 3 above)
Comment (zicanl-amazon) [reply sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:85]: fixed

Comment (rvasahu-amazon) [reply sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:90]: The diff currently shows this is not fixed, can you confirm?

File: sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb [hunk 5]
```diff
@@ -0,0 +1,1031 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1686b70e",
+   "metadata": {},
+   "source": [
+<<<<<<< HEAD
+    "## Install Development SDK\n",
+    "\n",
+    "As of 13-12-23, time series explainability (TSX) is in closed beta. The changes to the SDK to support it are not merged into the `sagemaker-python-sdk` repository. Instead, we install a version of the PySDK that does support these changes directly via a wheel file. "
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:90]: The diff currently shows this is not fixed, can you confirm?The diff currently shows this is not fixed, can you confirm?

Comment (rvasahu-amazon) [reply sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:90]: I cannot view the [notebook](https://github.com/zicanl-amazon/amazon-sagemaker-examples/blob/time-series/sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb). This looks like there are some issues caused by an unfinished git merge being committed.

Comment (zicanl-amazon) [reply sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:90]: checking

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb [hunk 6]
```diff
@@ -0,0 +1,2293 @@
...
+   "source": [
+    "!pip install mercury -q"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af541390-8a5c-4a5f-9cf0-d7a2b27d11b6",
+   "metadata": {},
+   "source": [
+    "## Install Development SDK\n",
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:152]: Notebook doesn't use development SDK anymore, please update this section.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb [hunk 7]
```diff
@@ -0,0 +1,2293 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "1985cfd8-7425-4008-bd30-0d11220f186f",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Summary\n",
+    "\n",
+    "This notebook is created to demonstrate the invocation of a SageMaker Clarify explainability job for a time series DeepAR forecasting model. Given a real model and a training dataset, a processingJob will be create to analyze the SHAP score for each feature attribute."
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:95]: >processingJob

Should be 'processing job'.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb [hunk 8]
```diff
@@ -0,0 +1,2293 @@
...
+    "\n",
+    "- `direction`: direction of explanation to be used. Available explanation types are `\"chronological\"`, `\"anti_chronological\"`, `\"bidirectional\"`. The cronological direction highlights the effect of older timesteps over more recent one, while the anti-chronological direction higlights the effect of timesteps closer to the forecasting. Bidirectional is a combination of the previous two modes. \n",
+    "- `granularity`: Granularity of explanation to be used. Available granularities are `\"timewise\"` and `\"fine_grained\"`. The first granularity is fast and computes the attribution of individual timesteps toward the forecast, not making distinctions of related time series. The fine-grained mode is slower, but computes an attribution for every timestep and every feature dynamic, distinguishing between related and target TS.\n",
+    "- `num_samples`: Number of samples to be used in the Asymmetric Shapley Value forecasting algorithm. Only applicable when using `\"fine_grained\"`  explanations. This represents the number of permutations sampled for computing the ASV. "
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:1103]: Can you add a section for the baseline to this markdown cell? You can probably copy the docstring from the SDK [here](https://github.com/aws/sagemaker-python-sdk/blob/49e09c3bbc9976b35d86c499884378a6b9cf5285/src/sagemaker/clarify.py#L1779) and modify it as necessary.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb [hunk 9]
```diff
@@ -0,0 +1,2293 @@
...
+    "    TimeSeriesJSONDatasetFormat,  # time series-specific dataset format\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e5297987",
+   "metadata": {},
+   "source": [
+    "### Set Configurations"
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:1054]: Not necessary but a nice-to-have: Can you link to (or include) documentation for content template and record template for time series? It's different from tabular explainability and we've had some users ask questions about this.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb (hunk 6 above)
Comment (zicanl-amazon) [reply sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:152]: Updated.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb [hunk 10]
```diff
@@ -0,0 +1,2293 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "ad0142ad-f709-4cb6-9b79-4ffa482341ca",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:1528]: As far as I can tell it doesn't look like this output using `mercury` is being displayed by GitHub. Perhaps it'd be best to swap to pretty printer?

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb (hunk 7 above)
Comment (zicanl-amazon) [reply sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:95]: I change it to explainability processing job.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb [hunk 11]
```diff
@@ -0,0 +1,2293 @@
...
+   "id": "e9dcec4e-7891-4703-a150-5a9d11d0748e",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'dataset_type': 'application/json',\n",
```
Comment (rvasahu-amazon) [inline sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:1455]: Format is al little strange due to spacing, can you change arguments to pretty printer to fix this?

**Review by rvasahu-amazon — CHANGES_REQUESTED** [review]
See comments.

File: sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb (hunk 5 above)
Comment (zicanl-amazon) [reply sagemaker-clarify/time_series_byom/time_series_bring_your_own_model.ipynb:90]: Interesting, I can't find it on my side. Will refresh the PR to see if it disappears.

File: sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb (hunk 8 above)
Comment (zicanl-amazon) [reply sagemaker-clarify/time_series_deepar/time_series_deepar.ipynb:1103]: Will do.

**Pull request closed without merging**